### DISCUSSION TITLE
Version Issue in integrating mobx-react in app

### DISCUSSION THREAD

**Discussion by AakankshaDahima** [mobxjs/mobx#2943] [Q&A]
I received error as for my app **entersekt-login-page** while running **_npm update_**

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: entersekt-login-page@1.0.0
npm ERR! Found: mobx-react@5.4.4
npm ERR! node_modules/mobx-react
npm ERR!   mobx-react@"5.4.4" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mobx-react@"^4.3.5" from bernie-core@0.41.4
npm ERR! node_modules/bernie-core
npm ERR!   peer bernie-core@">=0.24.7" from bernie-test-util@0.35.4
npm ERR!   node_modules/bernie-test-util
npm ERR!     dev bernie-test-util@"0.35.4" from the root project
npm ERR!   peer bernie-core@">=0.25.0" from @gc/global-controls-store@0.1.3
npm ERR!   node_modules/@gc/global-controls-store
npm ERR!     @gc/global-controls-store@"^0.1.3" from the root project
npm ERR!     1 more (bernie-test-util)
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
````

After upgrading mobx-react by **npm install mobx-react --save** I came up with same errors.
After installing mobx-react with specifically required version **npm i mobx-react@4.3.5**. Below error is coming up

```

npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: entersekt-login-page@1.0.0
npm ERR! Found: mobx@4.15.4
npm ERR! node_modules/mobx
npm ERR!   mobx@"4.15.4" from the root project
npm ERR!   peer mobx@"^3.4.1 || ^4.0.0" from bernie-core@0.41.4
npm ERR!   node_modules/bernie-core
npm ERR!     peer bernie-core@">=0.24.7" from bernie-test-util@0.35.4
npm ERR!     node_modules/bernie-test-util
npm ERR!       dev bernie-test-util@"0.35.4" from the root project
npm ERR!     peer bernie-core@">=0.25.0" from @gc/global-controls-store@0.1.3
npm ERR!     node_modules/@gc/global-controls-store
npm ERR!       @gc/global-controls-store@"^0.1.3" from the root project
npm ERR!       1 more (bernie-test-util)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mobx@"^2.6.3 || ^3.0.0" from mobx-react@4.3.5
npm ERR! node_modules/mobx-react
npm ERR!   mobx-react@"4.3.5" from the root project
npm ERR!   peer mobx-react@"^4.3.5" from bernie-test-util@0.35.4
npm ERR!   node_modules/bernie-test-util
npm ERR!     dev bernie-test-util@"0.35.4" from the root project
npm ERR!   2 more (@gc/global-controls-store, bernie-core)

```

Can someone help me out with suggestions around this?

**Comment by mweststrate** [comment]
The bernie-core project seems to want MobX 3 or 4. So best take it up with
the bernie project.

**Reply by AakankshaDahima** [reply]
> The bernie-core project seems to want MobX 3 or 4. So best take it up with the bernie project.
> […](#)
> On Tue, May 18, 2021 at 9:22 AM Aakanksha Dahima ***@***.***> wrote: I received error as for my app *entersekt-login-page* while running *npm update* npm ERR! code ERESOLVE npm ERR! ERESOLVE unable to resolve dependency tree npm ERR! npm ERR! While resolving: ***@***.*** npm ERR! Found: ***@***.*** npm ERR! node_modules/mobx-react npm ERR! mobx-react@"5.4.4" from the root project npm ERR! npm ERR! Could not resolve dependency: npm ERR! peer mobx-react@"^4.3.5" from ***@***.*** npm ERR! node_modules/bernie-core npm ERR! peer bernie-core@">=0.24.7" from ***@***.*** npm ERR! node_modules/bernie-test-util npm ERR! dev bernie-test-util@"0.35.4" from the root project npm ERR! peer bernie-core@">=0.25.0" from @***@***.*** npm ERR! ***@***.***/global-controls-store npm ERR! @gc/global-controls-store@"^0.1.3" from the root project npm ERR! 1 more (bernie-test-util) npm ERR! npm ERR! Fix the upstream dependency conflict, or retry After upgrading mobx-react by *npm install mobx-react --save* I came up with these errors. npm ERR! ERESOLVE unable to resolve dependency tree npm ERR! npm ERR! While resolving: ***@***.*** npm ERR! Found: ***@***.*** npm ERR! node_modules/mobx npm ERR! mobx@"4.15.4" from the root project npm ERR! peer mobx@"^3.4.1 || ^4.0.0" from ***@***.*** npm ERR! node_modules/bernie-core npm ERR! peer bernie-core@">=0.24.7" from ***@***.*** npm ERR! node_modules/bernie-test-util npm ERR! dev bernie-test-util@"0.35.4" from the root project npm ERR! peer bernie-core@">=0.25.0" from @***@***.*** npm ERR! ***@***.***/global-controls-store npm ERR! @gc/global-controls-store@"^0.1.3" from the root project npm ERR! 1 more (bernie-test-util) npm ERR! npm ERR! Could not resolve dependency: npm ERR! peer mobx@"^2.6.3 || ^3.0.0" from ***@***.*** npm ERR! node_modules/mobx-react npm ERR! mobx-react@"4.3.5" from the root project npm ERR! peer mobx-react@"^4.3.5" from ***@***.*** npm ERR! node_modules/bernie-test-util npm ERR! dev bernie-test-util@"0.35.4" from the root project npm ERR! 2 more ***@***.***/global-controls-store, bernie-core) Can someone help me out with suggestions around this? — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#2943>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAN4NBBS7TF6OGCEN4NVM5TTOIPTLANCNFSM45CAY7MQ> .

Hi @mweststrate , Even after installing mobx-react with specific version - 4, I am receiving that below error.

**Reply by mweststrate** [reply]
As the error suggest for mobx-react 4, you will need mobx 2 or 3. (their version numbers aren't related):

```
You got: 
npm ERR! Found: mobx@4.15.4

But your deps want:
npm ERR! Could not resolve dependency:
npm ERR! peer mobx@"^2.6.3 || ^3.0.0" from mobx-react@4.3.5
```

**Reply by AakankshaDahima** [reply]
> As the error suggest for mobx-react 4, you will need mobx 2 or 3. (their version numbers aren't related):
> 
> ```
> You got: 
> npm ERR! Found: mobx@4.15.4
> 
> But your deps want:
> npm ERR! Could not resolve dependency:
> npm ERR! peer mobx@"^2.6.3 || ^3.0.0" from mobx-react@4.3.5
> ```

While installing mobx 2 by `npm i mobx-react@2.0.0`, I am having this error, it keep on coming up according to versions.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: entersekt-login-page@1.0.0
npm ERR! Found: mobx-react@undefined
npm ERR! node_modules/mobx-react
npm ERR!   mobx-react@"2.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mobx-react@"^4.3.5" from bernie-core@0.41.4
npm ERR! node_modules/bernie-core
npm ERR!   peer bernie-core@">=0.24.7" from bernie-test-util@0.35.4
npm ERR!   node_modules/bernie-test-util
npm ERR!     dev bernie-test-util@"0.35.4" from the root project
npm ERR!   peer bernie-core@">=0.25.0" from @gc/global-controls-store@0.1.3
npm ERR!   node_modules/@gc/global-controls-store
npm ERR!     @gc/global-controls-store@"^0.1.3" from the root project
npm ERR!     1 more (bernie-test-util)
npm ERR! 

```
Can someone please suggest some command around this to resolve the issue?

**Comment by mweststrate** [comment]
***@***.***" sounds like something is broken on your side,
but beyond that I don't think we can be of much help from here, looks
some of your packages have inconsistent requirements of peers among
each other. So I'd check the bernie related packages + your mobx-react
version, and see what mobx and mobx-react versions they expect as
peers. Drawing a little graph on paper is probably the most helpful
here.
